pkg/virtctl/imageupload/imageupload.go

```diff
@@ -926,3 +946,56 @@ func handleEventErrors(client kubecli.KubevirtClient, pvcName, dvName, namespace
 
 	return nil
 }
+
+func createNewDataSource(client kubecli.KubevirtClient, name, namespace string) error {
+	ds := &v1beta1.DataSource{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: v1beta1.DataSourceSpec{
+			Source: v1beta1.DataSourceSource{
+				PVC: &v1beta1.DataVolumeSourcePVC{
+					Name:      name,
+					Namespace: namespace,
+				},
+			},
+		},
+	}
+	setDefaultInstancetypeLabels(ds.Labels)
+
+	_, err := client.CdiClient().CdiV1beta1().DataSources(namespace).Create(context.Background(), ds, metav1.CreateOptions{})
+	if err == nil {
+		fmt.Printf("Created a new DataSource %s/%s\n", namespace, name)
+	}
+	return err
+}
+
+func updateExistingDataSource(client kubecli.KubevirtClient, pvcName, pvcNamespace string, ds *cdiv1.DataSource) error {
+	setDefaultInstancetypeLabels(ds.Labels)
+
+	patchBytes, err := patch.GeneratePatchPayload(
+		patch.PatchOperation{
+			Op:    patch.PatchReplaceOp,
+			Path:  "/metadata/labels",
+			Value: ds.Labels,
+		},
+		patch.PatchOperation{
+			Op:   patch.PatchReplaceOp,
+			Path: "/spec/source/pvc",
+			Value: map[string]string{
+				"name":      pvcName,
+				"namespace": pvcNamespace,
+			},
+		},
+	)
+	if err != nil {
+		return err
+	}
+
+	_, err = client.CdiClient().CdiV1beta1().DataSources(ds.Namespace).Patch(context.Background(), ds.Name, types.JSONPatchType, patchBytes, metav1.PatchOptions{})
+	if err == nil {
```

Comment:
Same

Reply:
This line then becomes very very long. I know it's appreciated in Go, but it's hard to read it this way.